app/src/test/java/com/example/universitieslist/view/viewmodel/UniversityViewModelTest.kt

```diff
@@ -0,0 +1,84 @@
+package com.example.universitieslist.view.viewmodel
+
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
+import com.example.universitieslist.data.remote.repository.UniversityRepository
+import com.example.universitieslist.util.Constants
+import com.example.universitieslist.util.DispatcherHelper
+import com.example.universitieslist.util.mockUniversityResponses
+import io.mockk.MockKAnnotations
+import io.mockk.coEvery
+import io.mockk.every
+import io.mockk.impl.annotations.MockK
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.test.UnconfinedTestDispatcher
+import kotlinx.coroutines.test.advanceUntilIdle
+import kotlinx.coroutines.test.resetMain
+import kotlinx.coroutines.test.runTest
+import kotlinx.coroutines.test.setMain
+import org.junit.After
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertFalse
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+@OptIn(ExperimentalCoroutinesApi::class)
+class UniversityViewModelTest {
+
+    @get:Rule
+    val instantTaskExecutorRule = InstantTaskExecutorRule()
+
+    private val dispatcher = UnconfinedTestDispatcher()
+
+    private lateinit var viewModel: UniversityViewModel
+
+    @MockK
+    private lateinit var repository: UniversityRepository
+
+    @MockK
+    private lateinit var dispatcherHelper: DispatcherHelper
+
+    private val countryName = Constants.COUNTRY_NAME
+    private val expectedUniversityList = mockUniversityResponses
+
+    @Before
+    fun setUp() {
+        Dispatchers.setMain(dispatcher)
```

Comment:
You should consider writing a class for the dispatcher test class. Search for it online or I could help if you want. It would prevent duplication in each test class. You only need to add it as a rule in the test class and it will cater for dispatch and its switching.